libraries/botbuilder/tests/teamsActivityHandler.test.js

```diff
@@ -1526,6 +1526,221 @@ describe('TeamsActivityHandler', () => {
                 .catch(err => done(err));
         });
 
+        it('onTeamsTeamArchived routed activity', () => {
+            const bot = new TeamsActivityHandler();
+
+            let onTeamsTeamArchivedEventCalled = false;
+
+            const team = { id: 'team' };
+            const activity = createConvUpdateActivity({ team, eventType: 'teamArchived' });
+
+            bot.onConversationUpdate(async (context, next) => {
+                assert(context, 'context not found');
+                assert(next, 'next not found');
+                onConversationUpdateCalled = true;
+                await next();
+            });
+
+            bot.onTeamsTeamArchivedEvent(async (teamInfo, context, next) => {
+                assert(teamInfo, 'teamsInfo not found');
+                assert(context, 'context not found');
+                assert(next, 'next not found');
+                assert.strictEqual(teamInfo, team);
+                onTeamsTeamArchivedEventCalled = true;
+                await next();
+            });
+
+            bot.onDialog(async (context, next) => {
+                assert(context, 'context not found');
+                assert(next, 'next not found');
+                onDialogCalled = true;
+                await next();
+            });
+
+            const adapter = new TestAdapter(async context => {
+                await bot.run(context);
+            });
+
+            adapter.send(activity)
+                .then(() => {
+                    assert(onTeamsTeamArchivedEventCalled, 'onTeamsTeamArchivedEvent handler not called');
+                    assert(onConversationUpdateCalled, 'onConversationUpdate handler not called');
+                    assert(onDialogCalled, 'onDialog handler not called');
+                });

```

Comment:
Are any errors from the TeamsActivityHandler, TestAdapter and TestAdapter.send().then() being properly caught and surfaced?

Reply:
@stevengum  We'll add the catch clauses to all the rest of the tests. As soon as we have it ready, we'll let you know